lib/fluent/plugin_helper.rb

```diff
@@ -0,0 +1,36 @@
+#
+# Fluentd
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+require 'fluent/plugin_helper/event_emitter'
+require 'fluent/plugin_helper/thread'
+require 'fluent/plugin_helper/event_loop'
+require 'fluent/plugin_helper/timer'
+require 'fluent/plugin_helper/child_process'
+
+module Fluent
+  module PluginHelper
+    module Mixin
+      def self.included(mod)
+        mod.extend(Fluent::PluginHelper)
+      end
+    end
+
+    def helpers(*snake_case_symbols)
```

Comment:
> Plugin helper doesn't provide methods to override plugin methods (like format or others), and also doesn't overwrite any behavior of plugins by just included.

Does this depend on PluginHelper implementation or do you have any validation mechanizm?
Plugins use both with  `include XXX` code.
Seperating responsibility is good but it introduces complicated mechanizm.
And `Mixin` is too ruby-ish name so users may assume PluginHelper is good for providing development support like TypeConverter.